rocketpy/motors/Tank.py

```diff
@@ -0,0 +1,626 @@
+# -*- coding: utf-8 -*-
+
+__author__ = "Giovani Hidalgo Ceotto, Oscar Mauricio Prada Ramirez, João Lemes Gribel Soares, Mateus Stano, Pedro Henrique Marinho Bressan, Patrick Bales, Lakshman Peri, Gautam Yarramreddy, Curtis Hu, and William Bradford"
+__copyright__ = "Copyright 20XX, RocketPy Team"
+__license__ = "MIT"
+
+from abc import ABC, abstractmethod
+
+from rocketpy.Function import Function, funcify_method
+from rocketpy.tools import tuple_handler
+
+
+class Tank(ABC):
+    """Abstract Tank class that defines a tank object for a rocket motor, so
+    that it evaluates useful properties of the tank and its fluids, such as
+    mass, volume, fluid flow rate, center of mass, etc.
+    """
+
+    def __init__(self, name, geometry, flux_time, gas, liquid, discretize=100):
```

Comment:
So, maybe the name argument should be optional? 
In the AeroSurface's children classes, the name is always optional, since it is not so crucial to the simulation.

What do you think that is better?

Reply:
I prefer keeping it mandatory so that to encourage the user to label tanks in a explicit way. The naming is used in various exceptions and is fundamental to distinguish, for instance, which of the added Tanks overfilled.

Even if is not mandatory for simulation I think the parameter should always be set. In the future this might be revised if necessary.

Reply:
Agreed based on your reasons.
This is a different case of AeroSurface then.